src/react/useIsExtension.ts

```diff
@@ -0,0 +1,15 @@
+import { HandshakeStatus } from "src/types";
+import { useReplit } from "./useReplit";
+
+export default function useIsExtension(): undefined | boolean {
+  const { status } = useReplit();
+
+  switch (status) {
+    case HandshakeStatus.Ready:
+      return true;
+    case HandshakeStatus.Error:
+      return false;
```

Comment:
i wonder if we should explicitly tell the iframe when it is in fact not an extension, as opposed to inferring it from an error which could happen for other reasons (for example, there's other reasons the handshake could time out, like the main thread freezing up at the right time)

Reply:
Will merge this first, then maybe make some followup PRs